### PR TITLE
Adding Largest connected component analysis as a post-processing step

### PR DESCRIPTION
**Pull request by Geeks-Sid**
Adds connected component analysis

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change 
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [x] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ ] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation
- [x] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/CBICA/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CUDA10.2),[3](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CUDA11.3),[4](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-ROCm)]

### CODE REVIEW

File: testing/test_full.py [hunk 1]
```diff
@@ -1364,6 +1369,9 @@ def test_generic_preprocess_functions():
     input_tensor_image = sitk.GetImageFromArray(input_tensor.numpy())
     input_transformed = fill_holes(input_tensor_image)
 
+    input_tensor = torch.rand(1, 256, 256, 256) > 0.5
```
[the remark this answers is not in the document]
Comment (Geeks-Sid) [reply testing/test_full.py:1372]: Sure!, will add that

File: GANDLF/data/post_process/__init__.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
-from .morphology import torch_morphological, fill_holes
+from .morphology import fill_holes, cca
```
Comment (AlexanderGetka-cbica) [inline GANDLF/data/post_process/__init__.py:1]: This line still needs torch_morphological for the tests to work since they import it. (Assuming that test_full still needs torch_morphological -- if not just remove it from there)

**Comment by Geeks-Sid** [comment]
Anything pending for this @sarthakpati ?

**Comment by AlexanderGetka-cbica** [comment]
Can we resolve the conflicts in history.md?

File: GANDLF/data/post_process/morphology.py [hunk 3]
```diff
@@ -71,3 +73,30 @@ def fill_holes(input_image, params=None):
...
+    Returns:
+        torch.Tensor: The output image after morphological operations.
+    """
+    if isinstance(input_image, torch.Tensor):
+        seg = input_image.numpy()
+    elif isinstance(input_image, sitk.Image):
+        seg = sitk.GetArrayFromImage(input_image)
+    mask = seg != 0
+
+    connectivity = 3 if (params["connectivity"] is not None) else params["connectivity"]
```
[the remark this answers is not in the document]
Comment (Geeks-Sid) [reply GANDLF/data/post_process/morphology.py:95]: I don't understand this, could you explain in more detail?

[the remark this answers is not in the document]
Comment (Geeks-Sid) [reply GANDLF/data/post_process/morphology.py:95]: That makes sense.

File: GANDLF/data/post_process/morphology.py [hunk 4]
```diff
@@ -71,3 +73,30 @@ def fill_holes(input_image, params=None):
...
+    Returns:
+        torch.Tensor: The output image after morphological operations.
+    """
+    if isinstance(input_image, torch.Tensor):
+        seg = input_image.numpy()
+    elif isinstance(input_image, sitk.Image):
+        seg = sitk.GetArrayFromImage(input_image)
+    mask = seg != 0
+
+    connectivity = input_image.dim() - 1
```
[the remark this answers is not in the document]
Comment (Geeks-Sid) [reply GANDLF/data/post_process/morphology.py:95]: Why?

[the remark this answers is not in the document]
Comment (Geeks-Sid) [reply GANDLF/data/post_process/morphology.py:95]: but this is for masks, no?

**Review by AlexanderGetka-cbica — APPROVED** [review]
Looks good, guys.

**Comment by AlexanderGetka-cbica** [comment]
@sarthakpati you will need to resolve your previous review.
